ui/src/app/pages/profile/profile-page.tsx

```diff
@@ -419,8 +467,21 @@ export const ProfilePage = withUserProfile()(class extends React.Component<
             </a>
           </ProfileRegistrationStepStatus>}
         </div>
-
       </div>
+      {updatingSurvey && <Modal width={850}>
+        <DemographicSurvey
+            profile={profileEdits}
+            onCancelClick={() => {
+              this.setState({updatingSurvey: false});
+            }}
+            onSubmit={async(profileWithUpdatedDemographicSurvey, captchaToken) => {
+              this.setState({updatingSurvey: false});
```

Comment:
My understanding of this code is that the modal will close first, then the API request to save the updated profile will be sent? If we want to have a chance to handle an API error before losing the user's input, we should probably wait to hide the modal until we've received a successful API response.

Reply:
good call.

Reply:
alright, I'm going to have the parent handle this so it can do a bunch of z-indexes.